scripts/gpg-share.sh

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+
+ARCHIVE="keys.7z"
+
+
+# TODO: documentation
+log_error () {
+  echo -e "${0}: \e[91merror\e[0m: $1"
+}
+
+
+# TODO: documentation
+log_success () {
+  echo -e "${0}: \e[92msuccess\e[0m: $1"
+}
+
+
+# TODO: documentation
+log_warning () {
+  echo -e "${0}: \e[93mwarning\e[0m: $1"
+}
+
+
+# TODO: documentation
+log_info () {
+  echo -e "${0}: \e[96minfo\e[0m: $1"
+}
+
+
+# TODO: documentation
+check_file () {
+  [ -z "$1" ] && {
+    log_error "Please give a file to share as the first argument"
+    exit 1
+  }
+
+  [ ! -f "$1" ] && {
+    log_error "$1: No such file or directory"
+    exit 1
+  }
+}
+
+
+# TODO: documentation
+check_users() {
+  users=($@)
+
+  [ "${#users[@]}" -eq 0 ] && {
+    log_error "Please give at least one gpg id after the first argument"
+    exit 1
+  }
+
+  # check if each of the "user" is in the keyring
+  invalid=0
+  for user in "${users[@]}";
+  do
+      nb_ids=$(gpg --quiet --list-keys "$user" 2> /dev/null | grep "^uid" | wc -l)
+      case "$nb_ids" in
+        0 ) log_warning "'$user' not found in the keyring..."; invalid=1 ;;
+        1 )
+          id=$(gpg --quiet --list-keys "$user" | grep uid -B1 | head -n 1 | awk '{print $1}');
+          log_success "key $id found for '$user'"
+          ;;
+        * ) log_warning "too many ids found for '$user' in keyring ($nb_ids)"; invalid=1 ;;
+      esac
+  done
+  [ "$invalid" -eq 1 ] && {
+    log_error "one or more users are invalid for the keyring"
+    exit 1
+  }
+}
+
+
+file="$1"; shift 1
+check_file "$file"
+
+users=($@)
+check_users "${users[@]}"
+
+
+# TODO: documentation
+encrypt () {
+  for user in "${users[@]}";
+  do
+      log_info "encrypting '$file' for '$user'..."
+      gpg --verbose --recipient "$user" --encrypt --armor --output "$file.$user.asc" "$file"
+  done
+}
+
+
+# TODO: documentation
+archive () {
+  log_info "storing files in archive"
+  7zz a "$ARCHIVE" $(echo "${users[@]}" | tr ' ' '\n' | sed "s/\(.*\)/$file.\1.asc/")
```

Comment:
7zz? Same as 7z?
Could you add a dependency check for this?

Reply:
> 7zz? Same as 7z?
>
i do not know what `7z` is :thinking: 
i have
```bash
> pacman -Qo 7zz
/usr/bin/7zz is owned by 7-zip 22.01-2
```

> Could you add a dependency check for this?
>
yup, but what command should we use? `7zz`, `zip`, `7z`? :thinking:

Reply:
Hmmm
```bash
> pacman -Qo 7z
/usr/bin/7z is owned by p7zip 1:17.04-3
> pacman -Qo 7zz
error: No package owns 7zz
```

I think we should stick to `zip` because we don't need much compression and everyone has this one :+1:

Reply:
i do not have any `zip` command installed on my system :thinking:

Reply:
you tell me what command to use :relieved:

Reply:
Ok weird, what does `pacman -Qo zip` gives you? :open_mouth:

Reply:
> Ok weird, what does `pacman -Qo zip` gives you? open_mouth
>
```bash
> pacman -Qo zip
error: No package owns zip
```
it's just that `zip` is not installed by default on ArchLinux :+1:

Reply:
Okay that's a mess, after some discussion in private we'll go for `tar` as we hope that the one that most people will have

Reply:
@amtoine

Reply:
should be ok in https://github.com/iScsc/iscsc.fr/commit/f35fbd3ab5fb46daa8b7059a53b77be78eed13e1, https://github.com/iScsc/iscsc.fr/commit/8a6e5897524bb399430bdac1933d726bea390c1a and https://github.com/iScsc/iscsc.fr/commit/e8233a268f0cff16084d81abac7ec44c4661b092